src/commands/basemaps-github/make.cog.github.ts

```diff
@@ -159,6 +162,97 @@ export class MakeCogGithub {
+        // Alert if feature changed with no version bump.
+        return `🟥🟥🟥🟥 Feature Change Detected ${newLayer['lds:name']} - version: ${newLayer['lds:version']} features: ${newLayer['lds:feature_count']} (+${featureChange}) 🟥🟥🟥🟥`;
+      }
+
+      if (featureChange >= 0) {
+        // Add Features
+        return `🟦 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} (from: ${existingLayer['lds:version']}) features: ${newLayer['lds:feature_count']} (+${featureChange})`;
+      } else {
+        // Remove Features
+        return `🟧 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} (from: ${existingLayer['lds:version']}) features: ${newLayer['lds:feature_count']} (-${featureChange})`;
+      }
+    }
+
+    // Add new Layer
+    if (newLayer != null && existingLayer == null) {
+      return `🟩 ${newLayer['lds:name']} - version: ${newLayer['lds:version']} features: ${newLayer['lds:feature_count']}`;
+    }
+
+    // Remove Layer
+    if (newLayer == null && existingLayer != null) {
+      return `🟥 ${existingLayer['lds:name']} features: -${existingLayer['lds:feature_count']}`;
+    }
+
+    return;
+  }
+
+  /**
+   * Prepare and create pull request for the aerial tileset config
+   */
+  async diffVectorUpdate(layer: ConfigLayer, existingTileSet?: ConfigTileSetVector): Promise<string | undefined> {
+    const changes: (string | undefined)[] = [];
+    // Vector layer only support for 3857
+    if (layer[3857] == null) return;
+    const newCollectionPath = new URL('collection.json', layer[3857]).href;
+    const newCollection = await fsa.readJson<StacCollection>(newCollectionPath);
+    if (newCollection == null) throw new Error(`Failed to get target collection json from ${newCollectionPath}.`);
+    const ldsLayers = newCollection.links.filter((f) => f.rel === 'lds:layer');
+
+    // Log all the new inserts for new tileset
+    if (existingTileSet == null) {
+      changes.push(`New TileSet ts_${layer.name} with layer ${layer.name}.\n`);
+      for (const l of ldsLayers) {
+        changes.push(this.getVectorChanges(l, undefined));
+      }
+      return changes.filter((f) => f != null).join('\n');
+    }
+
+    // Compare the different of existing tileset, we usually only have one layers in the vector tiles, so the loop won't fetch very much
+    for (const l of existingTileSet.layers) {
+      if (l[3857] == null) continue;
+      if (l.name !== layer.name) continue;
+      changes.push(`Update for TileSet ${existingTileSet.id} layer ${layer.name}.`);
+      const existingCollectionPath = new URL('collection.json', l[3857]).href;
+      const existingCollection = await fsa.readJson<StacCollection>(existingCollectionPath);
+      if (existingCollection == null) {
+        throw new Error(`Failed to get target collection json from ${existingCollectionPath}.`);
+      }
+
+      // Prepare existing lds layers as map
+      const existingLdsLayers = new Map<unknown, StacLink>();
```

Comment:
```suggestion
      const existingLdsLayers = new Map<string, StacLink>();
```

do you then need to type the `(f) => f.rel === 'lds:layer') as StacLinkLds;` ?

Reply:
Yeah, we might need add a type for this other wise all the values are unknown.